src/validation.cpp

```diff
@@ -646,18 +646,20 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state, const C
             }
         }
 
-        // are the actual inputs available?
+        // This redundant check doesn't trigger the DoS code on purpose; if it did, it would make it easier
+        // for an attacker to attempt to split the network (Consensus::CheckTxInputs also checks this).
```

Comment:
Perhaps the new comment could be clearer, but the ```view.HaveInputs(tx)``` check below sets ```state.DoS(0, ...``` instead of ```state.DoS(100, ...``` like ```Consensus::CheckTxInputs```  would do.
Feel more than free to rephrase it in a way that would have been more clear to you.